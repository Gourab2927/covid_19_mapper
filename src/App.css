.App {
  text-align: center;
  background-color:rgb(81,82,82);
}

.Container {
  display: flex;
  flex-direction: row;  
  /* justify-content: center; */
}

.Map{
  position: absolute;
  width: 100%;
  height: 80%;
}

#navbar-item {
  position: flex;
  flex-direction: row;
  width: 25%;
  font-size:2vw;
}
