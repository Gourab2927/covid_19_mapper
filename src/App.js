import React, { Component } from "react";
import "./App.css";
import Header from "./components/navbar/navbar";
import MapContainer from "./components/map/map";
import ref_country_codes from "./components/assets/countries-lat-long.json";
import us_codes from "./components/assets/USlatlong.json";

class App extends Component {
  state = {
    countries: [],
    error: null,
    isLoaded: false,
    total: [],
  };

  componentDidMount() {
    Promise.all([
      fetch(
        "https://coronavirus-monitor.p.rapidapi.com/coronavirus/worldstat.php",
        {
          method: "GET",
          headers: {
            "x-rapidapi-host": "coronavirus-monitor.p.rapidapi.com",
            "x-rapidapi-key":
              "2bb49386fdmsh5daac6ca9add22ep1484a8jsn9816903163ef",
          },
        }
      ),
      fetch(
        "https://coronavirus-monitor.p.rapidapi.com/coronavirus/cases_by_country.php",
        {
          method: "GET",
          headers: {
            "x-rapidapi-host": "coronavirus-monitor.p.rapidapi.com",
            "x-rapidapi-key":
              "a04279196bmsh77bb3ff9e6f2e74p1f4d03jsn5bc0fbe15879",
          },
        }
      ),
      fetch("https://covid19-data.p.rapidapi.com/us", {
        method: "GET",
        headers: {
          "x-rapidapi-host": "covid19-data.p.rapidapi.com",
          "x-rapidapi-key":
            "2bb49386fdmsh5daac6ca9add22ep1484a8jsn9816903163ef",
        },
      }),
    ])

      .then(([res1, res2, res3]) => {
        return Promise.all([res1.json(), res2.json(), res3.json()]);
      })

      .then(([res1, res2, res3]) => {
        this.setState({
          countries: this.createCountry(res2.countries_stat, res3.list),
          total: this.updateTotal(res1),
        });
      });
  }

  createCountry(api_countries, states) {
    const countries = [];
    const usa = [];
    ref_country_codes.ref_country_codes.forEach((one) =>
      api_countries.forEach((two) => {
        if (two.country_name === "US") {
          usa.push(two);
        }
        if (one.country === two.country_name) {
          countries.push({
            country: two.country_name,
            recovered: two.total_recovered,
            deaths: two.deaths,
            confirmed: two.cases,
            center: { lat: one.latitude, lng: one.longitude },
            newCases: two.new_cases,
            newDeaths: two.new_deaths,
            activeCases: two.active_cases,
            criticalCases: two.serious_critical,
            perOneMillion: two.total_cases_per_1m_population,
          });
        }
      })
    );

    let a = this.updateUS(states, countries);

    return a;
  }

  updateUS(USA, countries) {
    var totalStates = {};
    let finalArray = [];

    var unique = USA.filter((v, i, a) => a.indexOf(v) === i);

    unique.forEach(function (d) {
      if (totalStates.hasOwnProperty(d.state)) {
        totalStates[d.state].deaths += d.deaths;
        totalStates[d.state].confirmed += d.confirmed;
      } else {
        totalStates[d.state] = {
          stateName: d.state,
          deaths: d.deaths,
          confirmed: d.confirmed
        };
      }
    });

    finalArray = Object.keys(totalStates).map((k) => totalStates[k]);

    us_codes.us_codes.forEach((state) =>
      finalArray.forEach((obj) => {
        if (obj.stateName === "Georgia") {
          obj.stateName = "Georgia, US";
        }
        if (obj.stateName === state.state) {
          countries.push({
            us: true,
            country: state.state,
            recovered: obj.recovered,
            deaths: obj.deaths,
            confirmed: obj.confirmed,
            center: { lat: state.latitude, lng: state.longitude },
          });
        }
      })
    );
    return countries;
  }

  updateTotal(totalArray) {
    var total = {}
    var newTotalCases = parseInt(totalArray["total_cases"].replace(/,/g, ""))
    var newTotalDeaths = parseInt(totalArray["total_deaths"].replace(/,/g, ""))
    var newTotalRecoveries = parseInt(totalArray["total_recovered"].replace(/,/g, ""))
    var activeCases = newTotalCases - newTotalDeaths - newTotalRecoveries

    total["total_cases"] = totalArray["total_cases"]
    total["total_deaths"] = totalArray["total_deaths"]
    total["total_recovered"] = totalArray["total_recovered"]
    total["new_cases"] = totalArray["new_cases"]
    total["new_deaths"] = totalArray["new_deaths"]
    total["statistic_taken_at"] = totalArray["statistic_taken_at"]
    total["active_cases"] = activeCases.toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ",")

    return total;
  }

  render() {
    return (
      <div className="App">
        <Header total={this.state.total} countries={this.state.countries} />
        <div className="Container">
          <MapContainer countries={this.state.countries} />
        </div>
      </div>
    );
  }
}

export default App;
