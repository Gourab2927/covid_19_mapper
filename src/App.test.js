import * as React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from './App';
import renderer from 'react-test-renderer';

describe("App Component", () => {  
  it('does not crash when loaded onto page', () => {
    shallow(<App />)
  })

  it('renders as expected, snapshot comparison', () => {
    const TextInputComponent = renderer.create(<App />).toJSON()
    expect(TextInputComponent).toMatchSnapshot()
  })
})

//  test('createCountry', () => {
//   const usStates = {'country.stat'{'country'{'United Kingdom'}}}
//   const countries = {{}}; 
//    expect(createCountry(usStates, countries)).toBe()
//  })


