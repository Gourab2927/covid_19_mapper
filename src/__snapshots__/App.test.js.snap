// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders as expected, snapshot comparison 1`] = `
<div
  className="App"
>
  <div
    id="app"
  >
    <nav
      className="navbar-dark bs-navbar-collapse navbar navbar-expand navbar-light fixed-top"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-outline-light"
            disabled={false}
            id="last_updated"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="spinner-grow spinner-grow-sm"
              role="status"
            />
          </button>
        </div>
      </div>
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-outline-info"
            disabled={false}
            id="last_updated"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="spinner-grow spinner-grow-sm"
              role="status"
            />
          </button>
        </div>
      </div>
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-outline-success"
            disabled={false}
            id="last_updated"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="spinner-grow spinner-grow-sm"
              role="status"
            />
          </button>
        </div>
      </div>
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-outline-danger"
            disabled={false}
            id="last_updated"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="spinner-grow spinner-grow-sm"
              role="status"
            />
          </button>
        </div>
      </div>
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-outline-warning"
            disabled={false}
            id="dropdown-menu-align-right"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="spinner-grow spinner-grow-sm"
              role="status"
            />
          </button>
        </div>
      </div>
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-outline-secondary"
            disabled={false}
            id="dropdown-menu-align-right"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="spinner-grow spinner-grow-sm"
              role="status"
            />
          </button>
        </div>
      </div>
    </nav>
    <div
      id="table"
    >
      <div>
        <div
          className="react-bootstrap-table"
        >
          <table
            className="table table-bordered"
          >
            <thead>
              <tr>
                <th
                  aria-label="Country sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  Country
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Total Cases sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  Total Cases
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Active Cases sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  Active Cases
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Recovered sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  Recovered
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Deaths sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  Deaths
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Critical Cases sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  Critical Cases
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Cases Per 1Million People sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  Cases Per 1Million People
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="C.F.R sortable"
                  className="sortable"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  tabIndex={0}
                >
                  C.F.R
                  <span
                    className="order"
                  >
                    <span
                      className="dropdown"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                    <span
                      className="dropup"
                    >
                      <span
                        className="caret"
                      />
                    </span>
                  </span>
                </th>
              </tr>
            </thead>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
